OpenEdXMobile/src/main/java/org/edx/mobile/deeplink/PushLinkManager.kt

```diff
@@ -0,0 +1,42 @@
+package org.edx.mobile.deeplink
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import com.google.firebase.messaging.RemoteMessage
+import de.greenrobot.event.EventBus
+import org.edx.mobile.event.PushLinkReceivedEvent
+import org.edx.mobile.logger.Logger
+
+object PushLinkManager {
+    val logger = Logger(this.javaClass)
+
+    fun checkAndReactIfReceivedFCMNotification(context: Context, intent: Intent?) {
+        val screenName = intent?.extras?.getString(DeepLink.Keys.SCREEN_NAME)
+        if (screenName != null && screenName.isNotEmpty()) {
+            // Received FCM background notification
+            didReceiveFCMBackgroundNotification(context, screenName, intent.extras)
+        }
+    }
+
+    private fun didReceiveFCMBackgroundNotification(context: Context, screenName: String, bundle: Bundle) {
+        DeepLinkManager.parseAndReact(context, DeepLink(screenName, bundle))
+    }
+
+    fun didReceiveFCMForegroundNotification(context: Context, remoteMessage: RemoteMessage?) {
+        if (remoteMessage == null || remoteMessage.notification == null) {
+            return
+        }
+        logger.debug("Message Notification Body: " + remoteMessage.notification!!.body)
```

Comment:
we can use '?' instated of '!!', '!!' will reproduce NullPointerException.

Ref: https://kotlinlang.org/docs/reference/null-safety.html#the--operator

Need updates in other places too.

Reply:
done